### DISCUSSION TITLE
Touch and Pointer Events Not Triggering Cropping on Mobile Devices in Fabric.js

### DISCUSSION THREAD

**Discussion by Arkit-Sutariya** [fabricjs/fabric.js#10137] [Q&A]
I am working with Fabric.js version 6.3.0 in an Angular 18.2.0 project and have implemented an image cropping feature that activates on a double-click event within the Fabric.js canvas. The cropping functionality works perfectly on desktop devices using the mouse event.

To extend this functionality to mobile devices, I added **touchstart** and **pointerdown** events. However, I encountered an issue where these events are not triggering when **enablePointerEvents: false** is set during canvas initialization. 

**Note** : I have also try with **enablePointerEvents: true**.

Here's the relevant code snippet:

`
import { FabricCropListener } from '../../js/latest-cropping.js';
import { Canvas, FabricImage, FabricObject } from 'fabric';

this.canvas = new Canvas('canvas', { hoverCursor: 'pointer',
    selection: true,
    selectionBorderColor: '#00c3f9',
    selectionColor: 'rgba(0, 195, 249, 0.2)',
    preserveObjectStacking: true,
    originX: 'center',
    originY: 'center',
    fireRightClick: true,
    perPixelTargetFind: false,
    enablePointerEvents: true
});

this.canvas.on({
    'mouse:dblclick': (e) => {
        consle.log("Double click fire");
        this.listener = new FabricCropListener(this.canvas, canvasSize);
         this.listener.crop()
        }
    },
    'touchstart': (e) => {
        console.log("Touch start event triggered");
    },
    'pointerdown': (e) => {
        console.log("Pointer down event triggered");
    },
});`

My goal is to enable seamless image cropping on mobile devices using **touchstart** and **pointerdown** events, but with **enablePointerEvents: false**  or **enablePointerEvents: true** set, these events do not trigger as expected.

I would appreciate any insights or solutions to help resolve this issue and ensure that cropping functionality works consistently across both desktop and mobile devices.

**Comment by rklf** [comment]
Having a similar issue. Have you been able to fix it?

**Reply by Arkit-Sutariya** [reply]
Hey @rklf 
I haven’t found any valid method so far, and I’ve paused work on this functionality.
However, the double-click event works fine on Android devices for now, but I haven’t tested it on iOS Safari yet.

**Reply by rklf** [reply]
Alright... thanks for you answer though.
